src/Reflex/Spider/Internal.hs

```diff
@@ -1378,20 +1405,51 @@ groupByHead = \case
       | otherwise -> (x, xs :| []) : l
 
 listsToForest :: Eq a => [[a]] -> Forest a
-listsToForest l = fmap (\(a, l') -> Node a $ listsToForest $ toList l') $ groupByHead $ catMaybes $ fmap nonEmpty l
+listsToForest lists = buildForest <$> groupByHead (mapMaybe nonEmpty lists)
+    where buildForest (a, lists') = Node a $ listsToForest $ toList lists'
+
+showStacks :: [[String]] -> String
+showStacks = drawForest . listsToForest . fmap (filterStack "Reflex.Spider.Internal")
```

Comment:
is this really going to be as useful as it could be without user code in stacks?  in fact to figure out what reflex is doing i have sometimes had to capture extra stacks when nodes are created, never mind filtering things out ...